src/main/java/io/cdap/plugin/snowflake/actions/loadunload/LoadUnloadConfig.java

```diff
@@ -0,0 +1,261 @@
+/*
+ * Copyright © 2020 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.snowflake.actions.loadunload;
+
+import com.google.common.base.Strings;
+import io.cdap.cdap.api.annotation.Description;
+import io.cdap.cdap.api.annotation.Macro;
+import io.cdap.cdap.api.annotation.Name;
+import io.cdap.cdap.etl.api.FailureCollector;
+import io.cdap.cdap.etl.api.validation.InvalidConfigPropertyException;
+import io.cdap.plugin.snowflake.common.BaseSnowflakeConfig;
+import java.util.stream.Stream;
+import javax.annotation.Nullable;
+
+/**
+ * Config for load and unload from Snowflake actions.
+ */
+public abstract class LoadUnloadConfig extends BaseSnowflakeConfig {
+  private static final String PROPERTY_FILE_FORMAT_FILTERING_POLICY = "fileFormatFilteringPolicy";
+  private static final String PROPERTY_CLOUD_PROVIDER = "cloudProvider";
+  private static final String PROPERTY_ENCRYPTION_TYPE = "encryptionType";
+  private static final String PROPERTY_FORMAT_TYPE = "formatType";
+  /**
+   * Cloud provider parameters
+   */
+  @Description("If true, plugin will use specified Cloud Provider Parameters.")
+  @Macro
+  private Boolean useCloudProviderParameters;
+
+  @Name(PROPERTY_CLOUD_PROVIDER)
+  @Description("Name of cloud provider. Possible values: GCP, AWS, Microsoft Azure")
```

Comment:
Since this is an optional field what is the default value?

Reply:
This config mainly lets our plugin decide which configs to show/hide in CDAP widget. 
Since for different providers they are different. Also it's used in code for the same purpose.

If it's not set, it's totally okay. It means that user does not want to use any provider specific params. He still can use any location like gcp://bucket, azure://... and aws://

Reply:
so nothing is provided can we make gcp default?

Reply:
Sorry it doesn't make sense to me. If nothing is provided this value is not used anywhere. User can provide any url like "gcp://..." or "aws://" and not even touch this config. 

This is only used to show specific parameters in certain cases. If it's "NONE" nothing is shown. Which makes sense, since we don't need to confuse user with extra params, that he does not need like FS encryption, keypair etc. These are only for specific use cases.

This makes plugin widget much cleaner and more usable, since most users will not need those extra params.